views/js/component/results.js

```diff
@@ -0,0 +1,251 @@
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+/**
+ * @author Jean-Sébastien Conan <jean-sebastien@taotesting.com>
+ */
+define([
+    'lodash',
+    'uri',
+    'core/eventifier',
+    'core/promise',
+    'ui/datatable'
+], function (_, uri, eventifier, Promise) {
+    'use strict';
+
+    /**
+     * Component that lists all the results entry points for a particular delivery
+     *
+     * @param {Object} config
+     * @param {String} config.classUri
+     * @param {String} config.dataUrl
+     * @param {Object} config.model
+     * @param {Object} areaBroker
+     * @param {Array} pluginFactories
+     * @returns {resultsList}
+     */
+    function resultsListFactory(config, areaBroker, pluginFactories) {
+        var resultsList;
+        var plugins = {};
+        var actions = [];
+        var classUri, model;
+
+        /**
+         * Calls a method from each plugins
+         *
+         * @param {String} method - the name of the method to run
+         * @returns {Promise} Resolved when all plugins are done
+         */
+        function pluginRun(method) {
+            var execStack = [];
+
+            _.forEach(plugins, function (plugin) {
+                if (_.isFunction(plugin[method])) {
+                    execStack.push(plugin[method]());
+                }
+            });
+
+            return Promise.all(execStack);
+        }
+
+        if (!_.isPlainObject(config)) {
+            throw new TypeError('The configuration is required');
+        }
+        if (!_.isString(config.classUri) || !config.classUri) {
+            throw new TypeError('The class URI is required');
+        }
+        if (!_.isString(config.dataUrl) || !config.dataUrl) {
+            throw new TypeError('The service URL is required');
+        }
+        if (!_.isPlainObject(config.model) && !_.isArray(config.model)) {
+            throw new TypeError('The data model is required');
+        }
+        if (!areaBroker || !_.isFunction(areaBroker.getListArea)) {
+            throw new TypeError('The areaBroker is required');
+        }
+
+        classUri = uri.decode(config.classUri);
+
+        /**
+         *
+         * @typedef {resultsList}
+         */
+        resultsList = eventifier({
+
+            /**
+             * Initializes the component
+             * @returns {resultsList} chains
```

Comment:
Because it is actually an application component, despite it is not using `ui/component`. It instanciates a datatable, have a lifecycle...